### PR TITLE
Make it possible to lambdify expressions that contain sums.

### PR DESCRIPTION
**Pull request by drufat**
For example, the lambdarepr of

```
Sum(x ** k, (k, k0, kN))
```

should be

```
(sum(x**k for k in range(k0, kN+1)))
```

### CODE REVIEW

**Comment by moorepants** [comment]
This looks good, but I'd like to see a test for lambdify if possible.

**Comment by drufat** [comment]
I converted the tests to use lambdify.

**Comment by moorepants** [comment]
+1 for me once the Travis tests pass.

For future improvements it would be cool to print a vectorized form for the 'numpy' backend.

**Comment by drufat** [comment]
Under the numpy backend there is a problem. The code as written above is using numpy.sum, rather than the intended __builtins__.sum which takes in a generator with items to be summed.  Having 
`(__builtins__.sum(x**k for k in range(k0, kN+1)))` as the template would fix that problem.

**Comment by drufat** [comment]
The lambda expression now uses `builtins.sum`. Also the compatible range function `sympy.core.compatibility.range` is passed so that the summation has constant memory complexity even in versions of python where range does not return a generator. Finally, tests are added to ensure that the vecorized version of the function in numpy works correctly.

**Comment by bjodah** [comment]
Look like you'll need a try/except ImportError block here to deal with Py2/3 compatibility (__builtin__ vs builtins respectively)

**Comment by drufat** [comment]
Apparently `sympy.core.compatibility` already provides `builtins` that is portable, so in the commit above I am just going to use that.

**Comment by bjodah** [comment]
Great, let's hope tests pass!

**Comment by drufat** [comment]
There were some merge conflicts in tests_numpy.py. These conflicts have now been resolved.

**Comment by asmeurer** [comment]
This doesn't handle multiple summations.

**Comment by asmeurer** [comment]
Otherwise it looks good. I don't know if there is a better way to do sums with numpy, though.

**Comment by drufat** [comment]
I added support for multiple summations.

File: sympy/printing/tests/test_lambdarepr.py [hunk 1]
```diff
@@ -145,5 +147,36 @@ def test_piecewise():
...
+
+    assert (lambdify((x, a, b), s)(2, 3, 8) ==
+            s.subs([(x, 2), (a, 3), (b, 8)]).doit())
+
+
+def test_multiple_sums():
+    s = Sum(i * x + j, (i, a, b), (j, c, d))
+
+    l = lambdarepr(s)
+    assert l == "(builtins.sum((builtins.sum(i*x + j for i in range(a, b+1))) " \
```
Comment (asmeurer) [inline sympy/printing/tests/test_lambdarepr.py]: `sum(i*x + j for i in range(a, b + 1) for j in range(c, d + 1))` should be the same but more efficient.

**Comment by drufat** [comment]
Ok. I modified the printer to give the more efficient version.

**Pull request merged**